jades/jades_validator.go

```diff
@@ -0,0 +1,125 @@
+package jades
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/fiware/VCVerifier/common"
+	"github.com/fiware/VCVerifier/logging"
+)
+
+const TOKEN_EXTRACTION_STRATEGY = "NONE"
+const DOCUMENT_NAME = "RemoteDocument"
+
+var ErrorBadResponse = errors.New("bad_response_from_validation_endpoint")
+var ErrorEmptyBodyResponse = errors.New("empty_body_response_from_validation_endpoint")
+var ErrorValidationServiceNotReady = errors.New("validation_service_not_ready")
+
+// Validator for JAdES(https://www.etsi.org/deliver/etsi_ts/119100_119199/11918201/01.01.01_60/ts_11918201v010101p.pdf) signatures
+type JAdESValidator interface {
+	ValidateSignature(signature string) (bool, error)
+}
+
+// Validator implementation, that uses an external validation service(based on https://github.com/esig/dss)
+type ExternalJAdESValidator struct {
+	HttpClient        common.HttpClient
+	ValidationAddress string
+	HealthAddress     string
+}
+
+// structs to be used with the dss-library
```

Comment:
Maybe nitpicking: From this service's point of view, it doesn't care about the dss-lib but rather about the https://github.com/wistefan/dss-validation-service

Reply:
It is not required to use my service;) The data structure comes from the library and every implementation following that lib can be used.

Reply:
The lib defines it as external;) https://github.com/esig/dss/blob/master/dss-validation-rest/src/main/java/eu/europa/esig/dss/ws/validation/rest/RestDocumentValidationServiceImpl.java